src/components/KCopy/KCopy.vue

```diff
@@ -0,0 +1,263 @@
+    return ''
+  }
+
+  // strip trailing colon from label if one exists
+  return props.badgeText.replace(/:$/, '')
+})
+
+const iconTitle = computed(() => {
+  if (props.badgeText) {
+    return `Copy ${badgeTextLabel.value}`
+  }
+  return 'Copy'
+})
+
+const truncateLimitText = computed((): string | null => props.truncate ? `${props.text.substring(0, props.truncationLimit) + '...'}` : null)
+
+// Computed for all dynamic classes
+const truncateStyles = computed((): string | null => props.truncate ? 'copy-element' : null)
+const badgeStyles = computed((): string | null => props.badge ? 'badge-styles' : null)
+const truncateContent = computed((): string | null => props.truncate ? 'truncate-content' : null)
+const useMono = computed((): string | null => props.badge ? null : 'monospace')
+
+const textFormat = computed(() => {
+  if (props.format === 'redacted') {
+    return '*****'
+  } else if (props.format === 'deleted') {
+    return `*${props.text.substring(0, 5)}`
+  }
+  return (props.truncate && props.truncationLimit && truncateLimitText.value) ? truncateLimitText.value.replace(/^"(.*)"$/, '$1') : props.text
+})
+
+const updateTooltipText = (msg: string): void => {
+  tooltipText.value = msg
+  setTimeout(() => {
+    tooltipText.value = props.copyTooltip
+  }, 1800)
+}
+
+const copyIdToClipboard = (executeCopy: (prop: string) => boolean) => {
+  if (!executeCopy(props.text)) {
+    if (hasSuccessTooltip.value) {
+      updateTooltipText('Failed to copy')
+    }
+
+    return
+  }
+
+  if (hasSuccessTooltip.value) {
+    updateTooltipText(props.successTooltip)
+    emit('copy', props.text)
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+.k-copy {
+  align-items: center;
+  display: flex;
+
+  &-badge-text {
+    margin-right: $kui-space-40;
+  }
+
+  .copy-element {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+
+    .truncate-content {
+      display: inline-block;
+      overflow: hidden;
+      text-overflow: ellipsis;
+      white-space: nowrap;
+    }
+  }
+
+  .badge-styles {
+    background-color: #f0f4fa;
+    border-color: #d7d8fe;
+    border-radius: var(--kui-border-radius-round, $kui-border-radius-round);
+    border-style: solid;
+    border-width: var(--kui-border-width-10, $kui-border-width-10);
+    color: #473cfb;
+    display: inline-flex;
+  }
+
+  .copy-container {
+    align-items: center;
+    cursor: pointer;
+    display: flex;
+    font-size: $kui-font-size-20;
+    line-height: $kui-line-height-20;
+    padding: $kui-space-10 $kui-space-30;
+    white-space: nowrap;
+  }
+
+  .monospace {
+    font-family: var(--kui-font-family-code, $kui-font-family-code);
+  }
+
+  .text-icon-wrapper {
+    align-items: center;
+    cursor: pointer;
+    display: flex;
+    margin-left: $kui-space-30;
+  }
+
+  .text-icon {
+    display: flex;
+  }
+
+  .copy-badge-text {
+    color: $kui-color-text-neutral;
+    font-size: $kui-font-size-20;
+    margin-right: $kui-space-20;
```

Comment:
```suggestion
    color: var(--kui-color-text-neutral, $kui-color-text-neutral);
    font-size: var(--kui-font-size-20, $kui-font-size-20);
    margin-right: var(--kui-space-20, $kui-space-20);
```

Reminder: please use CSS vars when using tokens in Kongponents